RP-69 핫딜 업로드 페이지

pushedRak opened the pull request:
## 🔥 관련 이슈

- Jira 이슈: [RP-69](https://cherrypick-devteam.atlassian.net/browse/RP-69)

<br/>

## 📝 변경사항

- [x] 이미지 업로드 방식 수정
 기존: Client에서 파일 배열로 관리하고 업로드 시 한 번에 Image API 호출
 수정: Client에서 이미지를 선택할 때마다 image API를 호출하여 image[]로 관리
- [x] 핫딜 상세 페이지에서 수정을 통해 페이지 진입 시 비어있는 데이터에 대해 수정 처리했습니다.

<br/>

## 📷 스크린샷 (선택)
>
> UI 변경 시 반드시 첨부

<br/>

## 📋 체크리스트

- [x] Jira 이슈와 연결함
- [x] 테스트 코드를 작성했거나, 충분히 테스트했음
- [x] PR 내용과 커밋 메시지에 이슈 키 포함

<br/>

---


🙌 봐주세요! :
> 


[RP-69]: https://cherrypick-devteam.atlassian.net/browse/RP-69?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

pushedRak committed:
Fix: 이미지 업로드 방식 수정 및 핫딜 수정 모드 관련 문제 수정

- 이미지 업로드 방식 수정
 - 기존: Client에서 파일 배열로 관리하고 업로드 시 한 번에 Image API 호출
 - 수정: Client에서 이미지를 선택할 때마다 image API를 호출하여 image[]로 관리

- 핫딜 상세 페이지에서 수정을 통해 페이지 진입 시 비어있는 데이터에 대해 수정 처리했습니다.

don9wan commented on src/services/apiDeal.ts:
```diff
@@ -44,6 +44,8 @@ export async function fetchRecommend(): Promise<FetchRecommendResponse> {
 export async function fetchDetailedDeal(id: string): Promise<DetailedDeal> {
     const result = await authRequest<DetailedDeal>(HttpMethod.GET, `/deal/${id}`);
+
+    console.log(result);
```
이거 테스트용 로그인지 확인 부탁드림다

pushedRak replied:
테스트용 로그 맞습니다

pushedRak committed:
Fix: 핫딜 업로드 로직 수정

- 핫딜 업로드 페이지 진입 시 dealId로 라우팅하지 않고 있던 문제 수정
- 핫딜 업로드 조회 API를 잘못된 방식으로 사용하고 있던 문제 수정 (이로 인해 discountName이 null로 들어오고 있었음)
- 수정 모드 진입 시 discount 방식들이 Modal 창에서 반영되지 않던 문제 수정

don9wan commented:
수고하셨슴니다
